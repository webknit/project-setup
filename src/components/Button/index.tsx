import React from 'react';
import './button.css';

interface ButtonProps {
    /**
     * Is this the principal call to action on the page?
     */
    primary?: boolean;
    /**
     * What background color to use
     */
    backgroundColor?: string;
    /**
     * How large should the button be?
     */
    size?: 'small' | 'medium' | 'large';
    /**
     * Button contents
     */
    label: string;
    /**
     * Optional click handler
     */
    onClick?: () => void;

    testId: string;
}

/**
 * Primary UI component for user interaction
 */
const Button = ({ primary = false, size = 'medium', backgroundColor, label, testId, ...props }: ButtonProps) => {
    const mode = primary ? 'storybook-button--primary' : 'storybook-button--secondary';
    console.log(mode);
    return (
        <button
            type="button"
            className={['storybook-button', `storybook-button--${size}`, mode].join(' ')}
            {...(testId ? { 'data-testid': testId } : {})}
            style={{ backgroundColor }}
            {...props}
        >
            {label}
        </button>
    );
};

export default Button;
