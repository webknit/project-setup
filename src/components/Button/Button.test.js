import renderer from 'react-test-renderer';
import Button from './index';

describe('<Button />', () => {
    describe('snapshot tests', () => {
        test('should render with no data', () => {
            const tree = renderer.create(<Button label="hello" />).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });

    describe('unit tests', () => {
        // test('should render ', () => {
        //     render(<Button />);
        // });
    });
});
