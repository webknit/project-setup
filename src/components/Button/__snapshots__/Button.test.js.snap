// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> snapshot tests should render with no data 1`] = `
<button
  className="storybook-button storybook-button--medium storybook-button--secondary"
  style={
    Object {
      "backgroundColor": undefined,
    }
  }
  type="button"
>
  hello
</button>
`;
